extensions/amp-story/0.1/amp-story-hint.js

```diff
@@ -0,0 +1,169 @@
+ * limitations under the License.
+ */
+
+import {renderSimpleTemplate} from './simple-template';
+import {dict} from '../../../src/utils/object';
+
+/** @private @const {!Array<!./simple-template.ElementDef>} */
+const NAVIGATION_HELP_OVERLAY = [
+  {
+    tag: 'div',
+    attrs: dict({'class': 'i-amphtml-story-navigation-help-overlay'}),
+    children: [
+      {
+        tag: 'div',
+        attrs: dict({'class': 'i-amphtml-story-navigation-help-section'
+            + ' prev-page'}),
+        children: [
+          {
+            tag: 'div',
+            attrs: dict({'class': 'i-amphtml-story-icons-container'}),
+            children: [
+              {
+                tag: 'div',
+                attrs: dict({'class': 'i-amphtml-story-hint-tap-icon'}),
+              },
+              {
+                tag: 'div',
+                attrs: dict({'class': 'i-amphtml-story-hint-tap-icon-text'}),
+              },
+            ],
+          },
+        ],
+      },
+      {
+        tag: 'div',
+        attrs: dict({'class': 'i-amphtml-story-navigation-help-section'
+            + ' next-page'}),
+        children: [
+          {
+            tag: 'div',
+            attrs: dict({'class': 'i-amphtml-story-icons-container'}),
+            children: [
+              {
+                tag: 'div',
+                attrs: dict({'class': 'i-amphtml-story-hint-tap-icon'}),
+              },
+              {
+                tag: 'div',
+                attrs: dict({'class': 'i-amphtml-story-hint-tap-icon-text'}),
+                text: 'Next page',
+              },
+            ],
+          },
+        ],
+      },
+    ],
+  },
+];
+
+/** @type {string} */
+const NAVIGATION_OVERLAY_CLASS = 'show-navigation-overlay';
+
+/** @type {string} */
+const FIRST_PAGE_OVERLAY_CLASS = 'show-first-page-overlay';
+
+/** @type {number} */
+const NAVIGATION_OVERLAY_TIMEOUT = 2000;
+
+/**
+ * User Hint Layer for <amp-story>.
+ */
+export class AmpStoryHint {
+
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @private {!Window} */
+    this.win_ = win;
+
+    /** @private {!Document} */
+    this.document_ = this.win_.document;
+
+    /** @private {?Element} */
+    this.hintContainer_ = null;
+
+    /** @private {?number} */
+    this.hintTimeout_ = null;
+  }
+
+  /**
+   * Builds the hint layer DOM.
+   * @return {!Element}
+   */
+  buildHintContainer() {
+    this.hintContainer_ = this.document_.createElement('aside');
+    this.hintContainer_.classList.add('i-amphtml-story-hint-container');
+    this.buildNavigationOverlay_();
+    return this.hintContainer_;
+  }
+
+  /**
+   * Builds navigation overlay DOM.
+   * @private
+   */
+  buildNavigationOverlay_() {
+    this.hintContainer_.appendChild(
+        renderSimpleTemplate(this.document_, NAVIGATION_HELP_OVERLAY));
+  }
+
+  /**
+   * Show navigation overlay DOM.
+   */
+  showNavigationOverlay() {
+    if (this.hintContainer_.classList.contains(NAVIGATION_OVERLAY_CLASS)) {
```

Comment:
Seems like this can be generalized?

```
showHint_(cssClassName, duration) {
  if (this.hintContainer_.classList.contains(FIRST_PAGE_OVERLAY_CLASS)) {
    return;
  }

  this.hideAllNavigationHint();
  this.hintContainer_.classList.add(FIRST_PAGE_OVERLAY_CLASS);

  this.hintTimeout_ = setTimeout(() => {
    this.hideAllNavigationHint();
  }, NAVIGATION_OVERLAY_TIMEOUT);
}
```

Then, the named versions (`showNavigationOverlay`, `showFirstPageHintOverlay`) just call the generic version with the appropriate args.  WDYT?

Reply:
So i've still left out the named versions of the functions to avoid maintaining a public set of class names or enum, but underneath they call a single function.

makes sense?

Reply:
Yep, sorry, this is what I meant!

Thanks 😄